e2e/start/basic/app/routes/isomorphic-fns.tsx

```diff
@@ -0,0 +1,61 @@
+import { createFileRoute } from '@tanstack/react-router'
+import { createIsomorphicFn, createServerFn } from '@tanstack/start'
+import { useState } from 'react'
+
+const getEnv = createIsomorphicFn()
+  .server(() => 'server')
+  .client(() => 'client')
+const getServerEnv = createServerFn().handler(() => getEnv())
+
+const getEcho = createIsomorphicFn()
+  .server((input: string) => 'server received ' + input)
+  .client((input) => 'client received ' + input)
+const getServerEcho = createServerFn()
+  .validator((input: string) => input)
+  .handler(({ data }) => getEcho(data))
+
+export const Route = createFileRoute('/isomorphic-fns')({
+  component: RouteComponent,
+})
+
+function RouteComponent() {
+  const [results, setResults] = useState<Partial<Record<string, string>>>()
+  async function handleClick() {
+    const env = getEnv()
+    const echo = getEcho('hello')
+    const [serverEnv, serverEcho] = await Promise.all([
+      getServerEnv(),
+      getServerEcho({ data: 'hello' }),
+    ])
+    setResults({ env, echo, serverEnv, serverEcho })
+  }
+  const { env, echo, serverEnv, serverEcho } = results || {}
+  return (
+    <div>
```

Comment:
A cool application for isomorphic functions can be working with hooks.
```ts
const useAccount = createIsomorphicFn()
  .server(() => [getStorage('currentUser'), noop])
  .client(() => useState(null))


function RouteComponent() {
  const [account, setAccount] = useAccount()
  // ...
}
```
This should work as expected